runtime/libia2/init.c

```diff
@@ -26,6 +26,10 @@ char *allocate_stack(int i) {
       exit(-1);
     }
   }
+#ifdef __aarch64__
+  /* Tag the allocated stack pointer so it is accessed with the right pkey */
+  stack = (char *)((uint64_t)stack | (uint64_t)i << 56);
```

Comment:
are we sure `stack`'s upper bits are initially zero or should we explicitly clear them out first?